projects/step-frontend/src/lib/modules/timeseries/uplot/uPlot.utils.ts

```diff
@@ -49,7 +59,7 @@ export class UPlotUtils {
       let lftDelta = cursorXVal - lftVal;
       let rgtDelta = rgtVal - cursorXVal;
 
-      hoveredIdx = lftDelta <= rgtDelta ? nonNullLft : nonNullRgt;
+      return lftDelta <= rgtDelta ? nonNullLft! : nonNullRgt!;
```

Comment:
Probably something you want to refactor later but just to highlight what we should improve for a better code style:

- shortened variables (Lft instead of Left)
- == instead of === (!==)
- null feels weird to me, are we sure that it is not just undefined?
- if statement without curly brackets "if (seriesData[i] != null) nonNullRgt = i;"
- statements that are difficult to read should get round brackets: "let rgtVal = (nonNullRgt == null) ? Infinity : xValues[nonNullRgt];"
- the bottom three let should be const

Reply:
I agree. I will refactor this. This is also 'part' of the library, that's why it looks like this.